kernel/src/engine/default/json.rs

```diff
@@ -119,8 +120,9 @@ impl<E: TaskExecutor> JsonHandler for DefaultJsonHandler<E> {
         &self,
         files: &[FileMeta],
         physical_schema: SchemaRef,
-        _predicate: Option<Expression>,
+        predicate: Option<Expression>,
     ) -> DeltaResult<FileDataReadResultIterator> {
+        debug!("Reading json files with predicate: {:?}", predicate);
```

Comment:
nit: might as well log the files too:
```suggestion
        debug!("Reading json files {files:?} with predicate: {predicate:?}");
```

Same below in parquet

Reply:
I moved the logging test to `scan::tests::test_scan_data`, making the logging in default unnecessary. Should I remove the logging in default, or keep it?

Reply:
Zach decided it's best to remove the unnecessary changes, so I'll mark this as resolved.